### PR TITLE
Allows disabling timeout from the client

### PR DESCRIPTION
**Pull request by TMSCH**
- Refactored timeout handling for `Request`s to be able to set them at the dispatch time.
- Added an `enableTimeouts` method on the `openyolo` instance to enable/disable globally timeouts.
- Removed timeout from the `SecureChannel` on the provider side.
- Wrote tests to cover `api.ts`

### CODE REVIEW

File: ts/api/api.ts [hunk 1]
```diff
@@ -313,6 +360,11 @@ class InitializeOnDemandApi implements OnDemandOpenYoloApi {
     this.reset();
   }
 
+  enableTimeouts(enable: boolean) {
```
Comment (iainmcgin) [inline ts/api/api.ts]: by convention, this should probably be "setTimeoutsEnabled".

File: ts/api/api.ts [hunk 2]
```diff
@@ -123,68 +123,107 @@ export interface OpenYoloApi {
 }
 
 /**
+ * Defines the different timeouts for every request.
+ */
+const TIMEOUTS: {[key: string]: number} = {
```
Comment (iainmcgin) [inline ts/api/api.ts]: const DEFAULT_TIMEOUTS_MS

Comment (TMSCH) [inline ts/api/api.ts]: Done.

File: ts/api/api.ts (hunk 1 above)
Comment (TMSCH) [inline ts/api/api.ts]: Done.

**Comment by TMSCH** [comment]
Thanks @iainmcgin!

**Pull request merged**